src/lsst/cm/tools/core/panda_utils.py

```diff
@@ -179,42 +179,45 @@ def determine_error_handling(dbi: DbInterface, errors_agg: dict, max_pct_failed:
     decision_results = []
     for key in errors_agg.keys():
         # for a given error, try to make a match
-        error_item = errors_agg[key]
+        error_items = errors_agg[key]
         pct_failed = max_pct_failed[key]
-        try:
-            error_match = dbi.match_error_type(error_item["panda_err_code"], error_item["diagnostic_message"])
-        except NameError:
-            error_match = False
-        # this section matches the error against things and gets
-        # information we need
-        # rescue, panda, intensity = get_error_recs(error_item)
-        # placeholders
-
-        # if there is no match, mark it as reviewable
-        if error_match is False:
-            temp_status = "failed_review"
-        # if this a known error critical enough that we need to pause
-        # then pause.
-        elif error_match["error_flavor"] == "critical":
-            temp_status = "failed_pause"
-        # if it is not a payload error nor critical, start a rescue
-        elif error_match["error_flavor"] != "payload":
-            temp_status = "failed_rescue"
-        # if the payload error is marked as rescueable, rescue
-        elif error_match["is_rescueable"] is True:
-            temp_status = "failed_rescue"
-        # is it supposed to be resolved?
-        elif error_match["is_resolved"] is True:
-            temp_status = "failed_review"
-        else:
-            # rework to count over the entire step and
-            # sum over the errors
-            max_intensity = error_match["max_intensity"]
-            if pct_failed >= max_intensity:
+        for error_item in error_items:
+            try:
+                error_match = dbi.match_error_type(
+                    error_item["panda_err_code"], error_item["diagnostic_message"]
+                )
+            except NameError:
+                error_match = False
+            # this section matches the error against things and gets
+            # information we need
+            # rescue, panda, intensity = get_error_recs(error_item)
+            # placeholders
+
+            # if there is no match, mark it as reviewable
+            if error_match in [False, None]:
+                temp_status = "failed_review"
+                # if this a known error critical enough that we need to pause
+                # then pause.
+            elif error_match.error_flavor is not None and error_match.error_flavor.name == "critical":
+                temp_status = "failed_pause"
+                # if it is not a payload error nor critical, start a rescue
+            elif error_match.error_flavor is not None and error_match.error_flavor.name != "payload":
+                temp_status = "failed_rescue"
+                # if the payload error is marked as rescueable, rescue
+            elif error_match.is_rescueable is True:
```

Comment:
This is not exclusive. We are going to run into a lot of `error_match.error_flavor is not None and error_match.error_flavor.name != "payload"` _and_ `error_match.is_rescueable is True`

Reply:
Talked with Orion via Slack, we think that this works the way we want.